idseq_dag/steps/run_star.py

```diff
@@ -1,12 +1,218 @@
+import multiprocessing
+import os
+
 from idseq_dag.engine.pipeline_step import PipelineStep
 import idseq_dag.util.command as command
+import idseq_dag.util.s3 as s3
+import idseq_dag.util.log as log
 
 
 class PipelineStepRunStar(PipelineStep):
+    max_input_reads = 75 * 1000 * 1000
+
     def run(self):
-        input_file_0 = self.input_files_local[0][0]
-        input_file_1 = self.input_files_local[0][1]
-        output_file_0 = self.output_files_local()[0]
-        output_file_1 = self.output_files_local()[1]
-        command.execute("gzip -dc %s |head -20 > %s" % (input_file_0, output_file_0))
-        command.execute("gzip -dc %s |head -20 > %s" % (input_file_1, output_file_1))
+        """Run STAR to filter out host reads."""
+        # Setup
+        input_files = self.input_files_local[0][0:2]
+        star_genome = self.additional_files["star_genome"]
+        ref_dir = self.ref_dir_local
+        scratch_dir = os.path.join(self.output_dir_local, "scratch")
+        total_counts_from_star = {}
+        output_files_local = self.output_files_local()
+        num_inputs = len(self.input_files[0])
+
+        # Check genome file. Ex: s3://...STAR_genome.tar -> /ref/STAR_genome
```

Comment:
actually, you don't need this any more. You can just do 

genome_dir = fetch_from_s3(self.additional_files["star_genome"],
                                   self.ref_dir_local,
                                   allow_s3mi=True,
                                   auto_untar=True)

I change the fetch_from_s3 code a bit, to make everything works seamlessly. you can just add it in this diff. in fetch_from_s3 do

+        if untar:
+            dst = dst[:-4] # Remove .tar